src/Master.cpp

```diff
@@ -59,27 +53,65 @@ void Master::handle_command_message(std::unique_ptr<Msg> CommandMessage) {
         "Kafka command doesn't contain timestamp, so using current time.");
   }
 
-  command_handler.tryToHandle(Message, TimeStamp);
+  handle_command(Message, TimeStamp);
 }
 
-void Master::handle_command(std::string const &Command) {
-  CommandHandler command_handler(getMainOpt(), this);
-  command_handler.tryToHandle(Command, getCurrentTimeStamp());
+nlohmann::json Master::parseCommand(std::string const &Command) {
+  try {
+    return nlohmann::json::parse(Command);
+  } catch (nlohmann::json::parse_error const &Error) {
+    throw std::runtime_error("Could not parse command JSON");
+  }
 }
 
-std::unique_ptr<StreamMaster> &
-Master::getStreamMasterForJobID(std::string const &JobID) {
-  for (auto &StreamMaster : StreamMasters) {
-    if (StreamMaster->getJobId() == JobID) {
-      return StreamMaster;
+void Master::handle_command(std::string const &Command,
+                            std::chrono::milliseconds TimeStamp) {
```

Comment:
I do not see that there is a good reason to move the timestamp around. It looks like it is only used in one location to me. Maybe get the current timestamp there?